Windows Updates hang with Cumulative Updates

When using packer to create Windows images with this update plugin, it always hangs when downloading the update.
If I log onto the system, I can see in the log files of this plugin, that the update is afterwards also installed, but the reboot is never executed. A manual reboot is not registered within packer (similar to the installation itself). This only seems to happen with the big cumulative updates.

My plugin call is simply:
"provisioners": [
    { "type": "windows-update" },
]

==> alicloud-ecs: Connected to WinRM!
==> alicloud-ecs: Uploading the Windows update elevated script...
==> alicloud-ecs: Uploading the Windows update check for reboot required elevated script...
==> alicloud-ecs: Uploading the Windows update script...
==> alicloud-ecs: Running Windows update...
    alicloud-ecs: Searching for Windows updates...
    alicloud-ecs: Found Windows update (2019-01-08; 1388.13 MB): 2019-01 Cumulative Update for Windows Server 2016 for x64-based Systems (KB4480961)
    alicloud-ecs: Downloading Windows updates (1 updates; 1388.13 MB)...

Unfortunately, my lack of go development experience has me stumped on how to recompile the suggested update by @JanWaller . 

('make' doesn't run. 'go build' does stuff until  update\assets.go:8:12: undefined: assets)

Any dummies guides on compiling plugins for packer on Windows? :)

Additionally, the suggestion of some output on screen sounds like a good idea to prevent timeouts. Any chance you'll roll this into an update @rgl 

Appreciate the help.

--- a/update/windows-update.ps1
+++ b/update/windows-update.ps1
@@ -177,10 +177,8 @@ for ($i = 0; $i -lt $searchResult.Updates.Count; ++$i) {
 
     $update.AcceptEula() | Out-Null
 
-    if (!$update.IsDownloaded) {
-        $updatesToDownloadSize += $update.MaxDownloadSize
-        $updatesToDownload.Add($update) | Out-Null
-    }
+    $updatesToDownloadSize += $update.MaxDownloadSize
+    $updatesToDownload.Add($update) | Out-Null
 
     $updatesToInstall.Add($update) | Out-Null
     if ($updatesToInstall.Count -ge $UpdateLimit) {

